src/Microsoft.AspNet.Mvc.Razor/Buffer/RazorValue.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Text.Encodings.Web;
+using Microsoft.AspNet.Html.Abstractions;
+
+namespace Microsoft.AspNet.Mvc.Razor.Buffer
+{
+    public struct RazorValue
+    {
+        private readonly bool _needsEncoding;
+
+        public RazorValue(string encoded, bool needsEncoding)
```

Comment:
We should just get rid of `needsEncoding` here and wrap the strings that **don't** need encoding. This was an experiment that was I was doing, and I'd prefer to continue it after implementing the buffer chunks.